### ISSUE TITLE
Reproducibility on higher versions of PyTorch + Torchvision + Cuda environment

### ISSUE THREAD

**Issue by songlin** [nianticlabs/monodepth2#414]
Hi,

Thanks for sharing the amazing work and still answering questions on GitHub after 3 years.

I have a problem reproducing the result for the Monocular model. Here is the result of the pretrained weights `mono_640x192.zip`
```
   abs_rel |   sq_rel |     rmse | rmse_log |       a1 |       a2 |       a3 | 
&   0.115  &   0.903  &   4.863  &   0.193  &   0.877  &   0.959  &   0.981  \\
```
It matches the number in your paper. But when I tried to train the model in my own environment, the best result I can get is 
```
   abs_rel |   sq_rel |     rmse | rmse_log |       a1 |       a2 |       a3 | 
&   0.115  &   0.926  &   4.907  &   0.194  &   0.877  &   0.958  &   0.980  \\
```
There is a noticeable gap for `sq_rel`, `rmse`.

My environment is Pytorch *1.9.0* + TorchVision *0.2.2* + Cuda *11.1* with RTX 2080TI. Before I start the training, I modified  Line *388* of `trainer.py` to explicitly specify `align_corers=True` to be compatible with the default behavior of older Torch versions like *0.4* you suggested in the installation.
```
# Line 388 of trainer.py, I added align_corners=True
outputs[("color", frame_id, scale)] = F.grid_sample(
                    inputs[("color", frame_id, source_scale)],
                    outputs[("sample", frame_id, scale)],
                    padding_mode="border", align_corners=True)
```

I also read the discussion [here](https://github.com/nianticlabs/monodepth2/issues/263) about reproducibility.
Following @[scott89](https://github.com/scott89) 's suggestion, i set
```
torch.backends.cudnn.enabled = False
```

Without these two changes, I obtained worse results
```
   abs_rel |   sq_rel |     rmse | rmse_log |       a1 |       a2 |       a3 | 
&   0.117  &   0.895  &   4.888  &   0.196  &   0.872  &   0.958  &   0.981  \\
```
This time, `sq_rel` and `rmse` are better, but `abs_rel` and `a1` are worse.

So, I was wondering is there anyone who successfully reproduces the results on higher Torch versions? And I suspect the difference could be caused by behavior changes of some torch&torchvision functions.

Thanks

**Comment by songlin** [comment]
By the way, all the results are from the best epoch of the last 5 epochs when doing the evaluation.

**Comment by dantkz** [comment]
Hi. Thanks for the interest.

It could be a number of things that contribute to final accuracy: [changes in ADAM behavior](https://github.com/pytorch/pytorch/pull/23737), changes in Cuda, changes in weight initialization, etc.

**Comment by songlin** [comment]
Thanks for the information. I am going to close this issue now.
